typing/ctype.ml

```diff
@@ -2521,8 +2537,16 @@ let add_gadt_equation uenv source destination =
     let expansion_scope =
       Int.max (Path.scope source) (get_equations_scope uenv)
     in
+    let type_origin =
+      match Env.find_type source env with
+      | decl -> type_origin decl
+      | exception Not_found -> assert false
```

Comment:
Do we want to document that `add_gadt_equation` requires the precondition  "`source` is available in `uenv`" (which is satisfied in all three call sites)?

Reply:
I kind of think that the default assumption is that a function which takes an env and a type should provide an env that contains the type. If you prefer I could default to `Origin_definition` or even the yet-to-exist `Origin_other` which I was considering adding for places like `reify` that don't yet have their own origin but aren't definitions either.

Reply:
I was more thinking about the `missing cmi` case: most of the other calls to `find_type` in `ctype.ml` handle this case directly, but `add_gadt_equation` can elide it since it has already been filtered by `is_intantiable`. 

I wouldn't mind an `Origin_other` for `reify`, but the `assert false` seems more self-documenting here.
(I am fine without adding a comment, this was mostly a review note for myself).